test/unit/reducers/projects.js

```diff
@@ -164,20 +172,44 @@ tap(initProjects({1: false}), projects =>
   )),
 );
 
+tap(initProjects({1: false}), projects =>
+  test('hideComponent', reducerTest(
+    reducer,
+    projects,
+    partial(hideComponent, '1', 'output', now),
+    projects.update('1', projectIn =>
+      projectIn.set('hiddenUIComponents', new Immutable.Set(['output'])).
+        set('updatedAt', now),
+    ),
+  )),
+);
+
+tap(initProjects({1: true}), projects =>
+  test('unhideComponent', reducerTest(
+    reducer,
+    projects,
+    partial(unhideComponent, '1', 'output', now),
+    projects.update('1', projectIn =>
+      projectIn.set('hiddenUIComponents', new Immutable.Set()).
+        set('updatedAt', now),
+    ),
+  )),
+);
+
 function initProjects(map = {}) {
   return reduce(map, (projectsIn, modified, key) => {
-    const projects = reducer(projectsIn, projectCreated(key));
+    let projects = reducer(projectsIn, projectCreated(key));
     if (modified) {
-      return reducer(
-        projects,
-        updateProjectSource(key, 'css', '', now),
-      );
+      projects = reducer(projects, updateProjectSource(key, 'css', '', now));
+      projects = reducer(projects, hideComponent(key, 'output', now));
```

Comment:
I would suggest just using what you have in the previous test, but reversing the order of the input/output. If you keep the initial stub modified (but without changing what that means) I think that should work perfectly.

May seem like a small point but the intent with passing a modified flag to the factory method here is *just* to mark it as having been modified; I’d prefer that the tests not make any other assumptions about what the actual modification is.